visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,39 +1,46 @@
-import { FileValidator, CCValidationResult } from "./util/fileValidator"
+import { validate, CCValidationResult } from "./util/fileValidator"
 import { AttributeTypes, CCFile, NameDataPair, BlacklistType, BlacklistItem } from "./codeCharta.model"
 import _ from "lodash"
 import { NodeDecorator } from "./util/nodeDecorator"
+import { OldAttributeTypes, ExportBlacklistType, ExportCCFile } from "./codeCharta.api.model"
 import { StoreService } from "./state/store.service"
 import { addFile, setSingle } from "./state/store/files/files.actions"
 import { getCCFiles } from "./model/files/files.helper"
+import { DialogService } from "./ui/dialog/dialog.service"
 
 export class CodeChartaService {
 	public static ROOT_NAME = "root"
 	public static ROOT_PATH = "/" + CodeChartaService.ROOT_NAME
 	public static readonly CC_FILE_EXTENSION = ".cc.json"
 
-	constructor(private storeService: StoreService) {}
+	constructor(private storeService: StoreService, private dialogService: DialogService) {}
```

Comment:
While it's correct TypeScript, I recommend to use as much JavaScript notation instead of TypeScript specific notation. JS does support private fields (`#privateProperty = "privateValue"`). It would then be transpiled for older browsers.

Reply:
Webpack should compile our Typescript code to JS and then transpile it with babel.
Correct me if I'm wrong here

Reply:
There's nothing wrong. TypeScript itself is for me just mostly a very powerful linter. It provides no actual benefit language wise besides that type declarations that I would really use (abstract classes are nice though).

It's just that more people know how to write JavaScript than TypeScript and TypeScript is always just a super set of JavaScript. Thus, if something already works with JavaScript itself, I would not use a different TypeScript notation (you also have more control over the transpiled code that way).